packages/babel-traverse/src/traverse-path.ts

```diff
@@ -0,0 +1,59 @@
+import TraversalContext from "./context";
+import type NodePath from "./path";
+import type { Node } from "@babel/types";
+import type { Visitor } from "./types";
+import { VISITOR_KEYS } from "@babel/types";
+import { explode } from "./visitors";
+
+export type TraversePathOptions = {
+  visitSelf?: boolean;
+};
+
+/**
+ * Traverse the given path, optionally visiting itself too
+ *
+ * @function
+ * @param path NodePath to be visited
+ * @param visitor The traverse options used to create a new traversal context
+ * @param state A user data storage provided as the second callback argument for traversal visitors
+ * @param opts User provided options for traversal
+ * @returns Whether the traversal stops early
+ */
+export function traversePath(
+  path: NodePath,
+  visitor: Visitor,
+  state?: any,
+  opts?: TraversePathOptions,
+): boolean {
+  const { type, scope } = path;
+
+  explode(visitor as Visitor);
+  const context = new TraversalContext(scope, visitor, state, path);
+
+  if (opts && opts.visitSelf) return context.visitQueue([path]);
+
+  const queue = buildQueue(path, type);
+  if (!queue.length) return false;
```

Comment:
I don't think we should return `false` if we do, for example, `booleanLiteralPath.traversePath(..., { visitSelf: false })` (boolean literals don't have children, so the queue would be empty). Are we using this return value for anything? If not, the function should return `void` (nothing).